### ISSUE TITLE
Understanding custom shaders in v4

### ISSUE THREAD

**Issue by gordyr** [pixijs/pixi.js#2957]
We have started looking at migrating our project to PIXI v4, and I've began toying with rewriting some of our custom PIXI extensions and modifications.

Among these modifications we have a whole host of custom shaders implemented for v3, that are designed to be used as rendering shaders, as opposed to ```PIXI.filters```, i.e by setting ```sprite.shader = customShader```.

This particular functionality is important to us for a variety of reasons.

I have rewritten one of our shaders in the new v4 filter format, but i'm unsure how to use it in the manner that we did previously, since now, sprite shaders are generated dynamically in order to provide multi-texture batching.

Currently, if I set the sprites ```sprite.shader``` property to our custom filter, it gets ignored and the standard multi-texture shader is generated.

So to summarise, how do we go about directly replacing rendering shader of a sprite now?  Is it even possible?

**Comment by ivanpopelyshev** [comment]
there's a plugin demo for it : https://github.com/pixijs/pixi-plugin-example . You can make your own old-style renderer that will accept filter shaders or so.

**Comment by gordyr** [comment]
Fantastic...  We need to rewrite several custom renderers for v4 anyway so that's no biggie.  Having the structure to hand, from the link you provided is great.  It should save us a fair bit of hacking away.

**Comment by ivanpopelyshev** [comment]
Look closer, it has some arguments you dont need. This thing was created to get rid of edge artifacts , you dont need some of them in your project.
